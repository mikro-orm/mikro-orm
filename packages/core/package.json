{
  "name": "@mikro-orm/core",
  "version": "6.1.12",
  "description": "TypeScript ORM for Node.js based on Data Mapper, Unit of Work and Identity Map patterns. Supports MongoDB, MySQL, PostgreSQL and SQLite databases as well as usage with vanilla JavaScript.",
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "typings": "dist/index.d.ts",
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.mjs"
      },
      "require": "./dist/index.js"
    }
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/mikro-orm/mikro-orm.git"
  },
  "funding": "https://github.com/sponsors/b4nan",
  "keywords": [
    "orm",
    "mongo",
    "mongodb",
    "mysql",
    "mariadb",
    "postgresql",
    "sqlite",
    "sqlite3",
    "ts",
    "typescript",
    "js",
    "javascript",
    "entity",
    "ddd",
    "mikro-orm",
    "unit-of-work",
    "data-mapper",
    "identity-map"
  ],
  "author": "Martin AdÃ¡mek",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/mikro-orm/mikro-orm/issues"
  },
  "homepage": "https://mikro-orm.io",
  "engines": {
    "node": ">= 18.12.0"
  },
  "scripts": {
    "build": "yarn clean && yarn compile && yarn copy && yarn run -T gen-esm-wrapper dist/index.js dist/index.mjs",
    "clean": "yarn run -T rimraf ./dist",
    "compile": "yarn run -T tsc -p tsconfig.build.json",
    "copy": "node ../../scripts/copy.mjs"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "dataloader": "2.2.2",
    "dotenv": "16.4.5",
    "esprima": "4.0.1",
    "fs-extra": "11.2.0",
    "globby": "11.1.0",
    "mikro-orm": "6.1.12",
    "reflect-metadata": "0.2.2"
  }
}
