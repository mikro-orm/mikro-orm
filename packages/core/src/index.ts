/**
 * @packageDocumentation
 * @module core
 */
/* istanbul ignore file */
export {
  Constructor, Dictionary, PrimaryKeyType, PrimaryKeyProp, Primary, IPrimaryKey, FilterQuery, IWrappedEntity, EntityName, EntityData, Highlighter,
  AnyEntity, EntityProperty, EntityMetadata, QBFilterQuery, PopulateOptions, Populate, Loaded, New, LoadedReference, LoadedCollection,
  GetRepository, EntityRepositoryType, MigrationObject, DeepPartial, PrimaryProperty, Cast, IsUnknown, EntityDictionary, EntityDTO, PlainObject,
} from './typings';
export * from './enums';
export * from './errors';
export * from './exceptions';
export * from './MikroORM';
export * from './entity';
export * from './events';
export * from './EntityManager';
export * from './unit-of-work';
export * from './utils';
export * from './hydration';
export * from './drivers';
export * from './connections';
export * from './platforms';
export * from './types';
export * from './naming-strategy';
export * from './metadata';
export * from './cache';
export * from './decorators';
