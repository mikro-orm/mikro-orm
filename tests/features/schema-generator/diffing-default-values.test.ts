import { Entity, MikroORM, PrimaryKey, Property } from '@mikro-orm/core';

export class Foo {

  @PrimaryKey()
  id!: number;

  @Property({ defaultRaw: "'test'" })
  bar0!: string;

  @Property({ default: 'test' })
  bar1!: string;

}

@Entity()
export class Foo1 extends Foo {

  @Property({ defaultRaw: 'now()' })
  bar2!: Date;

  @Property({ defaultRaw: 'now(6)', length: 6 })
  bar3!: Date;

}

@Entity()
export class Foo2 extends Foo {

  @Property({ defaultRaw: 'now()' })
  bar2!: Date;

  @Property({ defaultRaw: 'now()', length: 6 })
  bar3!: Date;

}

@Entity()
export class Foo3 extends Foo {

  @Property({ defaultRaw: 'now' })
  bar2!: Date;

}

describe('diffing default values (GH #2385)', () => {

  test('string defaults do not produce additional diffs [mysql]', async () => {
    const orm = await MikroORM.init({
      entities: [Foo1],
      dbName: 'mikro_orm_test_gh_2385',
      type: 'mysql',
      port: 3308,
    });
    await orm.getSchemaGenerator().refreshDatabase();
    expect(await orm.getSchemaGenerator().getCreateSchemaSQL()).toMatchSnapshot();
    await expect(orm.getSchemaGenerator().getUpdateSchemaSQL({ wrap: false })).resolves.toBe('');
    await orm.close();
  });

  test('string defaults do not produce additional diffs [mariadb]', async () => {
    const orm = await MikroORM.init({
      entities: [Foo1],
      dbName: 'mikro_orm_test_gh_2385',
      type: 'mariadb',
      port: 3309,
    });
    await orm.getSchemaGenerator().refreshDatabase();
    expect(await orm.getSchemaGenerator().getCreateSchemaSQL()).toMatchSnapshot();
    await expect(orm.getSchemaGenerator().getUpdateSchemaSQL({ wrap: false })).resolves.toBe('');
    await orm.close();
  });

  test('string defaults do not produce additional diffs [postgres]', async () => {
    const orm = await MikroORM.init({
      entities: [Foo2],
      dbName: 'mikro_orm_test_gh_2385',
      type: 'postgresql',
    });
    await orm.getSchemaGenerator().refreshDatabase();
    expect(await orm.getSchemaGenerator().getCreateSchemaSQL()).toMatchSnapshot();
    await expect(orm.getSchemaGenerator().getUpdateSchemaSQL({ wrap: false })).resolves.toBe('');
    await orm.close();
  });

  test('string defaults do not produce additional diffs [sqlite]', async () => {
    const orm = await MikroORM.init({
      entities: [Foo3],
      dbName: ':memory:',
      type: 'sqlite',
    });
    await orm.getSchemaGenerator().createSchema();
    expect(await orm.getSchemaGenerator().getCreateSchemaSQL()).toMatchSnapshot();
    await expect(orm.getSchemaGenerator().getUpdateSchemaSQL({ wrap: false })).resolves.toBe('');
    await orm.close();
  });

});
