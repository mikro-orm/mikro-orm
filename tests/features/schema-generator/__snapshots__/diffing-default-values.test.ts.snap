// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diffing default values (GH #2385) string defaults do not produce additional diffs [mariadb] 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

create table \`foo1\` (\`id\` int unsigned not null auto_increment primary key, \`bar0\` varchar(255) not null default 'test', \`bar1\` varchar(255) not null default 'test', \`bar2\` datetime not null default now(), \`bar3\` datetime(6) not null default now(6), \`metadata\` json not null default '{"value":42}') default character set utf8mb4 engine = InnoDB;

set foreign_key_checks = 1;
"
`;

exports[`diffing default values (GH #2385) string defaults do not produce additional diffs [mysql] 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

create table \`foo0\` (\`id\` int unsigned not null auto_increment primary key, \`bar0\` varchar(255) not null default 'test', \`bar1\` varchar(255) not null default 'test', \`bar2\` datetime not null default now(), \`bar3\` datetime(6) not null default now(6)) default character set utf8mb4 engine = InnoDB;

set foreign_key_checks = 1;
"
`;

exports[`diffing default values (GH #2385) string defaults do not produce additional diffs [postgres] 1`] = `
"set names 'utf8';
set session_replication_role = 'replica';

create table "foo2" ("id" serial primary key, "bar0" varchar(255) not null default 'test', "bar1" varchar(255) not null default 'test', "bar2" timestamptz not null default now(), "bar3" timestamptz(6) not null default now(), "metadata" jsonb not null default '{"value":42}');

set session_replication_role = 'origin';
"
`;

exports[`diffing default values (GH #2385) string defaults do not produce additional diffs [sqlite] 1`] = `
"pragma foreign_keys = off;

create table \`foo3\` (\`id\` integer not null primary key autoincrement, \`bar0\` text not null default 'test', \`bar1\` text not null default 'test', \`bar2\` datetime not null default now, \`metadata\` json not null default '{"value":43}');

pragma foreign_keys = on;
"
`;
